CNV/cnv-upgrade.sh

```diff
@@ -2,20 +2,14 @@
 
 set -ex
 
-OLD_CNV_VERSION="${OLD_CNV_VERSION:-2.0.0}"
-CNV_VERSION="${CNV_VERSION:-2.1.0}"
-OLD_CNV_CHANNEL="${OLD_CNV_VERSION:0:3}"
-CNV_CHANNEL="${CNV_VERSION:0:3}"
 TARGET_NAMESPACE="${TARGET_NAMESPACE:-openshift-cnv}"
 
-oc get sub hco-subscription -o yaml -n "${TARGET_NAMESPACE}" | sed "s/channel: ${OLD_CNV_CHANNEL}/channel: ${CNV_CHANNEL}/" | oc apply -n "${TARGET_NAMESPACE}" -f -
+oc get installplan -o yaml -n ${TARGET_NAMESPACE} $(oc get installplan -n ${TARGET_NAMESPACE} --no-headers | grep kubevirt-hyperconverged-operator.v2.1.0 | awk '{print $1}') | sed 's/approved: false/approved: true/' | oc apply -n openshift-cnv -f -
 
-oc get installplan -o yaml -n "${TARGET_NAMESPACE}" $(oc get installplan -n "${TARGET_NAMESPACE}" --no-headers | grep kubevirt-hyperconverged-operator.v"${CNV_VERSION}" | awk '{print $1}') | sed 's/approved: false/approved: true/' | oc apply -n "${TARGET_NAMESPACE}" -f -
-
-echo "Waiting until OLM replaces the ${OLD_CNV_VERSION} CSV"
-echo "This could take up to 10 minutes..."
-while [ -z "$(oc get csv -o'custom-columns=status:status.conditions[-1].phase,metadata:metadata.name' --no-headers | grep kubevirt-hyperconverged-operator.v${CNV_VERSION} | grep Succeeded)" ]; do
-    echo "Waiting for ${CNV_VERSION} CSV to be in 'Succeeded'..."
-    oc get csv -o'custom-columns=status:status.conditions[-1].phase,metadata:metadata.name' --no-headers | grep kubevirt-hyperconverged-operator
+echo "Waiting until update finishes"
+echo "This could take up some time..."
+oc get hco -n ${TARGET_NAMESPACE} hyperconverged-cluster -o=jsonpath='{range .status.conditions[*]}{.type}{"\t"}{.status}{"\t"}{.message}{"\n"}{end}'
```

Comment:
We have to be careful with our expectations when using conditions to imperatively measure when an upgrade completes.  Conditions are a declarative API that components use to pass status through an api server and eventually down to this client.  There is an opportunity for races.   As long as that's clear, I do think it provides a reasonable view of the upgrade.